spec/features/search_spec.rb

```diff
@@ -30,4 +30,153 @@
+      :customer,
+      name: "FooBar destroy_all: user",
+      email_subscriber: false)
+
+    visit admin_customers_path
+    fill_in :search, with: query
+    page.execute_script("$('.search').submit()")
+    expect(page).to have_content(customer.name)
+  end
+
+  scenario "admin searches a word into a model scope", :js do
+    searching_for = "Lua"
+    query = "scope:subscribed #{searching_for}"
+    subscribed_unmathed = create(
+      :customer,
+      name: "Dan Croak",
+      email_subscriber: true)
+    subscribed_matched = create(
+      :customer,
+      name: "#{searching_for} Miaus",
+      email_subscriber: true)
+    unsubscribed_matched = create(
+      :customer,
+      name: "#{searching_for} Doe",
+      email_subscriber: false)
+
+    visit admin_customers_path
+    fill_in :search, with: query
+    page.execute_script("$('.search').submit()")
+
+    page.within("tr.table__row", match: :first) do
+      expect(page).to have_content(subscribed_matched.name)
+    end
+    expect(page).not_to have_content(subscribed_unmathed.name)
+    expect(page).not_to have_content(unsubscribed_matched.name)
+  end
+
+  scenario "admin searches a word inside two model scopes", :js do
+    searching_for = "Lua"
+    query = "scope:subscribed scope:old #{searching_for}"
+    subscribed_unmathed = create(
+      :customer,
+      name: "Dan Croak",
+      email_subscriber: true)
+    unsubscribed_matched = create(
+      :customer,
+      name: "#{searching_for} Doe",
+      email_subscriber: false)
+    subscribed_match_but_new = create(
+      :customer,
+      created_at: 1.day.ago,
+      name: "#{searching_for} New",
+      email_subscriber: true)
+    subscribed_and_old_match = create(
+      :customer,
+      created_at: 5.years.ago,
+      name: "#{searching_for} Miaus",
+      email_subscriber: true)
+
+    visit admin_customers_path
+    fill_in :search, with: query
+    page.execute_script("$('.search').submit()")
+
+    page.within("tr.table__row", match: :first) do
+      expect(page).to have_content(subscribed_and_old_match.name)
+    end
+    expect(page).not_to have_content(subscribed_unmathed.name)
+    expect(page).not_to have_content(unsubscribed_matched.name)
+    expect(page).not_to have_content(subscribed_match_but_new.name)
+  end
+
+  scenario "admin clicks a scope button defined in an array", :js do
+    subscribed = create(
+      :customer,
+      name: "Lua Miaus",
+      email_subscriber: true)
+    unsubscribed = create(
+      :customer,
+      name: "John Doe",
+      email_subscriber: false)
+
+    visit admin_customers_path
+
+    # Included into the COLLECTION_SCOPES array of the CustomersDashboard
+    click_on "subscribed"
+
+    page.within("tr.table__row", match: :first) do
+      expect(page).to have_content(subscribed.name)
+    end
+    expect(page).not_to have_content(unsubscribed.name)
+  end
+
+  scenario "admin clicks a scope button defined in a hash", :js do
+    address_zip = '00000-9999'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.